design/0004-multicluster-phase-2.md

```diff
@@ -199,34 +201,62 @@ manifests for all of them:
 linkerd --context=source multicluster upgrade | kubectl --context=source apply -f -
 ```
 
-# Proposal 2: Endpointless Mirror Services
-
-TODO: flesh this out more
-
-* The target cluster gateway address is used in the Endpoints object for all
-  mirror services for a particular gateway and in the Endpoints object for the
-  gateway mirror.  Any time the gateway address changes, all of these objects
-  must be updated.
-* If we treated mirror services as a special case in the destination controller,
-  we could have the destination controller return the gateway address for
-  lookups against mirror services and not need to actually create or maintain
-  Endpoints objects.
-
-# Proposal 3: Use External Names when Gateway Address is a Hostname
-
-TODO: flesh this out more
-
-* When the gateway address is a hostname, we have the service mirror controller
-  resolve the hostname (and periodically re-resolve it) and use the resolved
-  IP address in mirror services and the gateway mirror.
-* In order to move this DNS resolution into the proxy we could update the
-  destination proxy-api to allow destination lookups to return hostnames.
-* By making the gateway mirror service into an external name service, we could
-  use the gateway address hostname as the gateway mirror external name.  The
-  destination controller would then be able to return this hostname for queries
-  on the gateway mirror and the proxy would be responsible for resolving DNS.
-* If the mirror services were also external name services OR if they were
-  special cased endpointless services as described in proposal 2, then the
-  destination controller could also return the gateway address hostname for
-  queries on the mirror services and the proxy would be responsible for
-  resolving DNS.
\ No newline at end of file
+## Unlinking
+
+To unlink a target cluster, we add an `unlink` command which outputs manifests
+for the `link` resource, the service mirror controller deployment, the cluster
+credentials, and all of the mirror services and gateway mirror for the given
+target cluster.  These can be piped to `kubectl delete`.
+
+```
+linkerd --context=source multicluster unlink --cluster-name=foobar | kubectl delete -f -
```

Comment:
What's the behavior if no name is specified?

Are all clusters unlinked? That would be bad 'default' behavior...

Should this require a list of cluster names, e.g. `ulink foobar`?

Reply:
Just as in the `link` command, the `--cluster-name` flag is mandatory.  

We also describe an explicit `--all-clusters` flag below.